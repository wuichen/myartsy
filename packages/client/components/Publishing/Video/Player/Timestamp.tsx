import { garamond } from "Assets/Fonts"
import React from "react"
import styled from "styled-components"
import { formatTime } from "../../Constants"

interface Props extends React.HTMLProps<HTMLDivElement> {
  duration: number
  currentTime: number
}

const TimestampComponent: React.SFC<Props> = props => {
  const { className, currentTime, duration } = props

  return (
    <div className={className}>
      {formatTime(currentTime)} / {formatTime(duration)}
    </div>
  )
}

export const Timestamp = styled(TimestampComponent)`
  ${garamond("s23")};
`
