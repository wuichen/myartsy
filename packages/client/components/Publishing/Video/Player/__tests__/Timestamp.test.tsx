import "jest-styled-components"
import React from "react"
import renderer from "react-test-renderer"
import { Timestamp } from "../Timestamp"

describe("Timestamp", () => {
  it("matches the snapshot", () => {
    const timestamp = renderer
      .create(<Timestamp duration={5000} currentTime={2000} />)
      .toJSON()
    expect(timestamp).toMatchSnapshot()
  })
})
