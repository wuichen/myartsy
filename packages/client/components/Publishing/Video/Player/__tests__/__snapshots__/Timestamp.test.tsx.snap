// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timestamp matches the snapshot 1`] = `
.c0 {
  font-family: 'Adobe Garamond W08','adobe-garamond-pro','AGaramondPro-Regular','Times New Roman',Times,serif;
  font-size: 23px;
  line-height: 1.5em;
  -webkit-font-smoothing: antialiased;
}

<div
  className="c0"
>
  33:20
   / 
  23:20
</div>
`;
